### ISSUE TITLE
[BUG] Caddy is unable to get ssl cert and unable to get nginx reverse proxy to work

### ISSUE THREAD

**Issue by jaselee0905** [PhlexPlexico/G5API#270]
**Describe the bug**
I have my host name set to my domain name which  I used a cname to direct to my public ip but caddy continues to fail to get any kind of ssl certificate. I have nginx proxy manager on a different machine on the same network that could possibly be interfering but it shouldn't be. 
![image](https://github.com/PhlexPlexico/G5API/assets/41090702/fc90970e-f757-4968-9e2a-73fd9d4a226b)

I tried to then use nginx to host it and used a reverse proxy to connect to it using the server config file from the readme for nginx but I get a 500 Internal error, I've to change values in the conf file but I either get 500 or a 502 error
`##
# Default server configuration
#
server {
        listen 80 default_server;
        listen [::]:80 default_server;

        # SSL configuration
        #
        #listen 443 ssl default_server;
        #listen [::]:443 default_server;

        # This is where the build for G5V resides, update this to wherever you need.
        root /var/www/html;

        # Add index.php to the list if you are using PHP
        index index.html index.htm index.nginx-debian.html;

        # Change this to whichever hostname you have, such as get5.phlexplexi.co
        server_name lan.mythril.ca;

        # This is the main block that was added. Currently the API runs on port 3301. You may adjust to what you need it to if you changed ports in the config.
        location /api/ {
                proxy_pass http://192.168.2.51:3301/;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection 'upgrade';
                proxy_set_header Host $host;
                proxy_cache_bypass $http_upgrade;
        }

        # IF YOU'RE USING THE DOCKER IMAGE, THIS WILL HAVE TO BE UPDATED IN THE DOCKER IMAGE ITSELF. Do not put this on your reverse proxy.
        location / {
                # First attempt to serve request as file, then
                # as directory, then fall back to displaying a 404.
                try_files $uri /index.html;
        }

}`

This is all running through portainer on Ubuntu, I am use to docker but the networking stuff is new to me and I have been struggling to get this to work for too long

**Comment by PhlexPlexico** [comment]
Hey,

Chances are if you're using a reverse proxy on your network, it could be interfering. However, it looks like that it's showing 404 when trying to obtain the SSL cert via Caddy. When creating the docker compose file did you also create the network adapter for everything? `docker network create -d bridge get5`. Also, please ensure that all your URLs in the docker compose are set to HTTPS and not HTTP, as it looks like the error is claiming you're trying to get to an HTTP address.

https://github.com/PhlexPlexico/G5API/blob/f2a633d6d81dce52b7d19e31fe276814588e5578/docker-compose.yml#L60-L61

If you could, would you mind pasting your docker compose with any sensitive information removed? (passwords, steam api keys, etc).

**Comment by jaselee0905** [comment]
Yes I created the network adapter, This is the one I used with caddy, for the one without caddy I used pretty much exactly this except without all the caddy stuff and I built my own G5V front end to have the server conf above
`version: "3.7"

services:
  redis:
    image: redis:6
    command: redis-server --requirepass 
    container_name: redis2
    networks:
      - get5
    restart: always

  get5db:
    image: yobasystems/alpine-mariadb
    container_name: get5db2
    restart: always
    networks:
      - get5
    environment:
      - MYSQL_ROOT_PASSWORD=
      - MYSQL_DATABASE=get5
      - MYSQL_USER=get5
      - MYSQL_PASSWORD=
      - MYSQL_CHARSET=utf8mb4
      - MYSQL_COLLATION=utf8mb4_general_ci

  caddy:
    image: lucaslorentz/caddy-docker-proxy:ci-alpine
    container_name: caddy-reverse-proxy2
    restart: unless-stopped
    networks:
      - get5
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    ports:
      - 80:80
      - 443:443
    environment:
      - CADDY_INGRESS_NETWORKS=get5

  g5api:
    image: ghcr.io/phlexplexico/g5api:latest
    depends_on:
      - get5db
    container_name: G5API2
    networks:
      - get5
    labels:
      caddy: lan.mythril.ca
      caddy.handle_path: /api/*
      caddy.handle_path.0_reverse_proxy: "{{upstreams 3301}}"
    volumes:
      - ./public:/Get5API/public
    environment:
      - NODE_ENV=production
      - PORT=3301
      - DBKEY=
      - STEAMAPIKEY=
      - HOSTNAME=https://lan.mythril.ca
      - SHAREDSECRET=
      - CLIENTHOME=https://lan.mythril.ca
      - APIURL=https://lan.mythril.ca/api
      - SQLUSER=get5
      - SQLPASSWORD=123123
      - SQLPORT=3306
      - DATABASE=get5
      - SQLHOST=get5db
      - ADMINS=76561198049897893
      - SUPERADMINS=76561198049897893
      - REDISURL=redis://:super_secure_password_change_me@redis:6379
      - REDISTTL=86400
      - USEREDIS=true
      - UPLOADDEMOS=true
      - LOCALLOGINS=false
    restart: always

  g5v:
    image: ghcr.io/phlexplexico/g5v:latest
    depends_on:
      - g5api
    container_name: G5V-Front-End2
    networks:
      - get5
    restart: always
    labels:
      caddy: lan.mythril.ca
      caddy.reverse_proxy: "{{upstreams}}"

networks:
  get5:
    external: true`

**Comment by PhlexPlexico** [comment]
I'm assuming this ` - REDISURL=redis://:super_secure_password_change_me@redis:6379` was included to whatever your redis password was set to up on the 6th line? ` command: redis-server --requirepass` but this password was just removed?

**Comment by jaselee0905** [comment]
Yes the redis password were matching I hadn't changed it yet because I was just trying to get it to work first, I removed it from line 6 while just quickly removing everything

**Comment by PhlexPlexico** [comment]
Are you able to check the logs of the other containers, specifically G5API and G5V to ensure they're running? If it's giving a 404 I worry that your instance of G5V may not be running, or closed out. I believe if you wanted to set something up on NGINX there is an additional rewrite rule as well to ensure that all `/api` calls are rewritten as `/api/`, but I'm unsure as to where that is, and I think you can google the rewrite rules fairly easily. 


As for the Caddy implementation, have you tried temporarily disabling your NGINX reverse proxy and see if you run into the same issue?

**Comment by jaselee0905** [comment]
They are both running here are the logs for both of them. I also know that the API is working since if I manually go to the /api/ addresses it works and does the auth and logout so it seems to be an issue with the front end. 
  
[_G5V-Front-End_logs.txt](https://github.com/PhlexPlexico/G5API/files/14528712/_G5V-Front-End_logs.txt)

[_G5API_logs.txt](https://github.com/PhlexPlexico/G5API/files/14528711/_G5API_logs.txt)

For the caddy version I have tried it with my proxy manager off and I had the same issue

**Comment by jaselee0905** [comment]
I have made the modifications the server conf I posted in the first post is the one I'm using it's near the bottom

**Comment by PhlexPlexico** [comment]
Ah, well then. That may be the issue lol. Try pulling the front end from GHCR and don't make any modifications to it and see if it works?

**Comment by jaselee0905** [comment]
I'm sorry what is GHCR? 
Also it seems like If I don't make modifications to the conf then the backend doesn't seem to work properly, I get error  whenever it goes to the API like described in the readme, but with the modifications the API works but not the front end

**Comment by PhlexPlexico** [comment]
GHCR = GitHub Container Registry - basically where the compiled image is. Personally this is the setup that I have running. I've made zero changes to anything in the containers, the only change I've made to the docker compose file is remove the port 443 from the caddy image, and set port 80 to be port 9000 externally, since it's not in use.

Since I have a reverse proxy container already (I'm hosting multiple things), my Caddy file looks like the following for G5API and G5V:

```json
get5.phlexplexi.co {
   reverse_proxy G5V-Front-End
   request_body {
     max_size 256MB
   }
  redir /api /api/
  route /api/* {
    uri strip_prefix /api
    reverse_proxy http://G5API:3301
  }
}
```
Realistically, if you're using Docker to set this up you shouldn't have to make any changes in the containers themselves. Worst case with making no changes in the G5V container would mean if you went to a route directly, like `/matches/1` without going throught he main page, it would 404. 

You mentioned you already have an NGINX reverse proxy running. What you could try is the same thing as me, and adjust the ports of the caddy image in the docker compose, and convert the above JSON to an NGINX conf for your subdomain, and see if that would work? Also, apologies for the long winded ness of this - I haven't exactly done setup or worked with this project extensively for a few years now, so I'm also a bit rusty in trying to set things up :)

**Comment by jaselee0905** [comment]
Ok I figured it out thank you for pointing me in the right direction, it turns out nginx proxy managers custom location does not work properly, I wrote a custom line for it instead and everything seems to work now.

Thank you for your help I really appreciate it

**Comment by PhlexPlexico** [comment]
No problem! Glad to hear it's now working as intended :)

**Comment by loof2736** [comment]
@jaselee0905 what was the custom line you used to get it working with nginx proxy manager?

**Comment by jaselee0905** [comment]
@loof2736 Hey sorry, I really wish I could help more but I can't remember exactly what I did and my nginx proxy manager has bricked itself since and I had to reset it. But if I remember right instead of using custom locations in nginx proxy manger I did it manually in the advanced tab instead.

**Comment by loof2736** [comment]
For anybody else with this problem I got it working in this discussion: https://github.com/PhlexPlexico/G5API/discussions/298
